src/ports/referral-server/component.ts

```diff
@@ -0,0 +1,77 @@
+import { AppComponents } from '../../types'
+import { IReferralServerComponent, ReferralError } from './types'
+
+export async function createReferralServerComponent({
```

Comment:
Would you mind renaming the name of the component to be named in terms of what they're about instead of a service? This seems better to not attach the what's doing to the server, which can always change.